src/vs/workbench/contrib/terminalContrib/suggest/common/terminalSuggestConfiguration.ts

```diff
@@ -212,11 +212,20 @@ export function registerTerminalSuggestProvidersConfiguration(availableProviders
 		default: false
 	};
 
+
 	if (availableProviders) {
 		for (const providerId of availableProviders) {
 			if (providerId in defaultValue) {
 				continue;
+			} else if (providerId.startsWith('lsp')) {
+				providersProperties[providerId] ??= {
+					type: 'boolean',
+					description: localize(`suggest.provider.lsp.${providerId}.description`, "Enable or disable the Language Server Protocol (LSP) based provider. This provides enhanced completion support through language servers."),
+					default: product.quality !== 'stable'
+				};
+				continue;
 			}
+
```

Comment:
@andyleejordan I'm trying to test this in the powershell extension, does the extension not analyze all schemes? We need the `vscode-terminal` scheme to support completions in LSP for this feature.

@anthonykim1 or am I doing something wrong here? There's no `lsp` provider here after installing the pwsh ext:

<img width="1168" height="796" alt="image" src="https://github.com/user-attachments/assets/a954f12e-8545-46f5-8beb-fe074a74998d" />

Reply:
@Tyriar Can you try opening pwsh terminal after activating powershell extension (I think the easiest way is opening ps1 file), and then the pwsh extension would show up as a provider:
<img width="1516" height="744" alt="Screenshot 2025-08-07 at 8 02 20 AM" src="https://github.com/user-attachments/assets/9a80a41e-6d76-4109-8a5b-1feb144a06a1" />